lib/galaxy/managers/base.py

```diff
@@ -387,6 +387,27 @@ def list(self, filters=None, order_by=None, limit=None, offset=None, **kwargs):
         items = self._apply_fn_filters_gen(items, fn_filters)
         return list(self._apply_fn_limit_offset_gen(items, limit, offset))
 
+    def count(self, filters=None, **kwargs):
+        """
+        Returns the number of objects matching the given filters.
+
+        If the filters include functional filters, the count is performed by getting all objects
+        and then applying the functional filters which may impact performance.
+        """
+        # TODO: requires case sensitivity fix from https://github.com/galaxyproject/galaxy/pull/16036
+        orm_filters, fn_filters = self._split_filters(filters)
+        query = self.query(filters=orm_filters, **kwargs)
+        if not fn_filters:
+            # if no fn_filtering required, we can use count
+            return query.count()
+
+        # TODO: raise an error instead to avoid performance issues?
+
+        # fn filters will change the number of items
```

Comment:
The list of known function filters is `annotation`, `genome_build`, `data_type` and `community_rating`. `community_rating` we should drop, and the rest seems like it could be converted into orm or column filters relatively easily ... or we should at least give it a try.

In the meantime I think just raising an invalid request exception seems fine.

Reply:
Thank you! I will do that